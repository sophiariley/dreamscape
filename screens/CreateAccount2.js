// The second page the user interacts with when creating a new account

import React, { useState} from "react";
import { StyleSheet, Text, TextInput, View, KeyboardAvoidingView, TouchableOpacity, ScrollView, Dimensions, Alert } from "react-native";
import {AntDesign} from 'react-native-vector-icons';
//import { getDatabase, ref, set } from "firebase/database";
import { doc, setDoc, collection, addDoc } from "firebase/firestore"; 
import { db } from "../firebase-config";

const CreateAccount2 = ({route, navigation}) => {
    const { firstName, lastName, email } = route.params;
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [password2, setPassword2] = useState('');
    const [userID, setUserID] = useState(''); //CAN BE USED TO PASS UID THROUGH INSTEAD OF FINDING IT LATER
    //const res = db.collection('users');

    function createUser(firstName, lastName, email, username, password) {
        const runit = async () => await addDoc(collection(db, "users"), {
       firstName: firstName,
       lastName: lastName,
       email: email,
       username: username,
       password: password
     })
     .then(function(docRef) {
        console.log("Create Acct UID: ", docRef.id);
        setUserID(docRef.id);
     });
     runit();
    }

    const windowHeight = Dimensions.get('window').height;
    const windowWidth = Dimensions.get('window').width;

     // Alert user if they left empty fields
    const emptyAlert = () =>
    Alert.alert(
        'Empty Field',
        'Please make sure all fields are filled out!',
        [
        {text: 'Ok', style: 'cancel'}
        ]
    );

    // Alerts user if input in confirmed password box doesn't match that in password box
    const wrongPasswordAlert = () => 
    Alert.alert(
        'Error',
        'Your password does\'t match.',
        [
        {text: 'Ok', style: 'cancel'}
        ]
    );

    const doBoth = () => {
        createUser(firstName, lastName, email, username, password); 
        navigation.navigate('Home', {
            username: username,
            password: password,
        });
    }
    
    async function auth(username, password, password2) {
        let verified = [false, ''];
        if (username=='' || password=='' || password2=='') { emptyAlert(); return verified; } 
        if (password!=password2) { wrongPasswordAlert(); return verified; }

        createUser(firstName, lastName, email, username, password); 
        const q = query(collection(db, "users"), where("username", "==", username), where("password", "==", password));
        const querySnapshot = await getDocs(q);
        querySnapshot.forEach((doc) => {
            console.log(doc.id, " => ", doc.data());
            verified = [true, doc.id];
        });
        return verified;
    }

    return (
        <View style={styles.container}>
            <ScrollView style={{flex:1}}>
                <View style={{width:windowWidth, height:windowHeight }}>
                    <Text style={styles.createNewAccount}>Create New Account</Text>

                    <View style={styles.textContainer}>
                        <Text style={styles.text}>Enter Username</Text>
                            {/* Input box for user to enter their desired username */}
                            <TextInput 
                                style={styles.inputText} 
                                placeholder=""
                                value={username}
                                onChangeText={text => setUsername(text)}
                            />

                        <Text style={styles.text}>Enter Password</Text>
                            {/* Input box for user to enter their desired password */}
                            <TextInput 
                                style={styles.inputText} 
                                placeholder=""
                                secureTextEntry //hides password
                                value={password}
                                onChangeText={text => setPassword(text)}
                            />

                        <Text style={styles.text}>Re-enter Password</Text>
                            {/* Input box for user to confirm their password */}
                            <TextInput 
                                style={styles.inputText} 
                                placeholder=""
                                secureTextEntry //hides password
                                value={password2}
                                onChangeText={text => setPassword2(text)}
                            />
                    </View>

                    {/* Next button, which saves the user's input and navigates to HomeScreen */}
                    <View style={styles.nextButtonContainer}>
                        <TouchableOpacity 
                            onPress={() => { 
                                auth(username, password).then(
                                    function(value) {
                                        if (value[0]) {
                                            navigation.navigate('Home', {
                                                userID: value[1],
                                                username: username,
                                                show: true,
                                            })
                                        }
                                        else {
                                            // wrongAlert();
                                        } 
                                    }
                                );
                            }}
                            //;username=='' || password=='' || password2=='' ? emptyAlert() : password!=password2 ? wrongPasswordAlert() : doBoth() } }
                            // change to Login later

                            style={styles.nextButton}
                        > 
                            <Text style={styles.next}>Next</Text>
                            <AntDesign name='arrowright' size={20} color='#F6F6F6'/>
                        </TouchableOpacity>
                    </View>

                    {/* Previous button, which navigates to CreateAccount1Screen */}
                    <View style={styles.prevButtonContainer}>
                        <TouchableOpacity
                            onPress={() => navigation.navigate('Create Account 1')}
                            style={styles.prevButton}
                        > 
                            <AntDesign name='arrowleft' size={20} color='#F6F6F6'/>
                            <Text style={styles.prev}>Previous</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </ScrollView>
        </View>
    )
}

export default CreateAccount2;

// Styles used for this screen:s
const styles = StyleSheet.create({
    // A general container for components
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },

    // A container for the input boxes and their respection text
    textContainer: {
        top: 200,
        position: 'absolute',
        alignSelf: 'center'
    },

    // A container for the "Next" button
    nextButtonContainer: {
        width: '70%',
        height: '20%',
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row',
        bottom: 80,
        position: 'absolute',
        alignSelf: 'center'
    },

    // A container for the "Previous" button
    prevButtonContainer: {
        width: '70%',
        height: '20%',
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row',
        position: 'absolute',
        bottom: 10,
        alignSelf: 'center'
    },

    // The styling for the "Create New Account" text
    createNewAccount: {
        color: '#F8C98A',
        fontSize: 35,
        top: 100,
        position: 'absolute',
        alignSelf: 'center'
    },

    // Styling for the input text
    inputText: {
        fontSize: 20,
        backgroundColor: '#DADADA',
        borderRadius: 5,
        width: 350,
        height: '20%',
        paddingLeft: 10
    },

    // Styling for the text about the input boxes
    text: {
        color: '#3A6496',
        fontSize: 14,
        paddingTop: 10,
        paddingBottom: 5
    }, 

    // Styling for the "Next" button
    nextButton: {
        backgroundColor: '#D28A8E', // pink color
        width: '100%',
        padding: 15,
        borderRadius: 5,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row'
    },

    // Styling for the "Next" text
    next: {
        fontSize: 20,
        color: '#F6F6F6',
        paddingRight: 10,
    },

    // Styling for the "Previous" button
    prevButton: {
        backgroundColor: '#3A6496', // blue color
        width: '100%',
        padding: 15,
        borderRadius: 5,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row'
    },

    // Styling for the "Previous" text
    prev: {
        fontSize: 20,
        color: '#F6F6F6',
        paddingLeft: 10,
    },
})