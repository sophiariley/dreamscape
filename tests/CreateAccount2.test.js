import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native'; 
import {create} from 'react-test-renderer';
import CreateAccount2 from './CreateAccount2';


// basic rendering test

test('renders correctly across screens', () => {
  const tree = create(<CreateAccount2 />);
  expect(tree).toMatchSnapshot();
});


// Testing username box

const USERNAME_TEXT = 'username';

describe('Testing username input', () => {
    it('should find the input box via testId', () => {
        const testIdName = 'username';
        const {getByTestId} = render(<CreateAccount2 />);
        const foundButton = getByTestId(testIdName);
        expect(foundButton).toBeTruthy();
    });

    it('Input box is pressable', () => {
        const page = render(<CreateAccount2/>);
        const usernameInput = page.getByTestId('username');
        fireEvent.press(usernameInput);
    });

    test('Input box should allow for input', () => {
      render(<CreateAccount2/>);
      fireEvent.changeText(screen.getByTestId('username'), USERNAME_TEXT)
    })
});

// Testing password box

const PASSWORD_TEXT = 'password';

describe('Testing password input', () => {
    it('should find the input box via testId', () => {
        const testIdName = 'password';
        const {getByTestId} = render(<CreateAccount2 />);
        const foundButton = getByTestId(testIdName);
        expect(foundButton).toBeTruthy();
    });

    it('Input box is pressable', () => {
        const page = render(<CreateAccount2/>);
        const passwordInput = page.getByTestId('password');
        fireEvent.press(passwordInput);
    });

    test('Input box should allow for input', () => {
      render(<CreateAccount2/>);
      fireEvent.changeText(screen.getByTestId('password'), PASSWORD_TEXT)
    })
});

// Testing password box

const REPPASSWORD_TEXT = 'password';

describe('Testing password input', () => {
    it('should find the input box via testId', () => {
        const testIdName = 'repPassword';
        const {getByTestId} = render(<CreateAccount2 />);
        const foundButton = getByTestId(testIdName);
        expect(foundButton).toBeTruthy();
    });

    it('Input box is pressable', () => {
        const page = render(<CreateAccount2/>);
        const passwordInput = page.getByTestId('repPassword');
        fireEvent.press(passwordInput);
    });

    test('Input box should allow for input', () => {
      render(<CreateAccount2/>);
      fireEvent.changeText(screen.getByTestId('repPassword'), REPPASSWORD_TEXT)
    })
});


// Testing next button

describe('Testing next button', () => {
    it('should find next button via testId', () => {
    const testIdName = 'nextButton';
    const {getByTestId} = render(<CreateAccount2 />);
    const foundButton = getByTestId(testIdName);
    expect(foundButton).toBeTruthy();
    });
  
    it('Next button is pressable', () => {
      const navigation = {navigate: () =>{}}
      const page = render(<CreateAccount2 navigation={navigation}/>);
      const nextButton = page.getByTestId('nextButton');
      fireEvent.press(nextButton);
    });
  })


// Testing back button

describe('Testing back button', () => {
    it('should find back button via testId', () => {
    const testIdName = 'backButton';
    const {getByTestId} = render(<CreateAccount2 />);
    const foundButton = getByTestId(testIdName);
    expect(foundButton).toBeTruthy();
    });
  
    it('Back button is pressable', () => {
      const navigation = {navigate: () =>{}}
      const page = render(<CreateAccount2 navigation={navigation}/>);
      const backButton = page.getByTestId('backButton');
      fireEvent.press(backButton);
    });

    it('Back button navigates to login screen', () => {
        const navigation = {navigate: () =>{}}
        const page = render(<CreateAccount2 navigation={navigation}/>);
        jest.spyOn(navigation, 'navigate')
        const backButton = page.getByTestId('backButton');
        fireEvent.press(backButton);
        expect(navigation.navigate).toHaveBeenCalledWith("Create Account 1")
    });
})
// Testing last name box

// const LASTNAME_TEXT = 'last name';

// describe('Testing username input', () => {
//     it('should find the input box via testId', () => {
//         const testIdName = 'lastName';
//         const {getByTestId} = render(<CreateAccount1 />);
//         const foundButton = getByTestId(testIdName);
//         expect(foundButton).toBeTruthy();
//     });

//     it('Input box is pressable', () => {
//         const page = render(<CreateAccount1/>);
//         const usernameInput = page.getByTestId('lastName');
//         fireEvent.press(usernameInput);
//     });

//     test('Input box should allow for input', () => {
//       render(<CreateAccount1/>);
//       fireEvent.changeText(screen.getByTestId('lastName'), LASTNAME_TEXT)
//     })
// });


// // Testing email box

// const EMAIL_TEXT = 'email';

// describe('Testing username input', () => {
//     it('should find the input box via testId', () => {
//         const testIdName = 'email';
//         const {getByTestId} = render(<CreateAccount1 />);
//         const foundButton = getByTestId(testIdName);
//         expect(foundButton).toBeTruthy();
//     });

//     it('Input box is pressable', () => {
//         const page = render(<CreateAccount1/>);
//         const usernameInput = page.getByTestId('email');
//         fireEvent.press(usernameInput);
//     });

//     test('Input box should allow for input', () => {
//       render(<CreateAccount1/>);
//       fireEvent.changeText(screen.getByTestId('email'), EMAIL_TEXT)
//     })
// });


// // Testing confirm email box

// const CONFIRMEMAIL_TEXT = 'email';

// describe('Testing username input', () => {
//     it('should find the input box via testId', () => {
//         const testIdName = 'confirmEmail';
//         const {getByTestId} = render(<CreateAccount1 />);
//         const foundButton = getByTestId(testIdName);
//         expect(foundButton).toBeTruthy();
//     });

//     it('Input box is pressable', () => {
//         const page = render(<CreateAccount1/>);
//         const usernameInput = page.getByTestId('confirmEmail');
//         fireEvent.press(usernameInput);
//     });

//     test('Input box should allow for input', () => {
//       render(<CreateAccount1/>);
//       fireEvent.changeText(screen.getByTestId('confirmEmail'), CONFIRMEMAIL_TEXT)
//     })
// });


// // Testing next button

// describe('Testing next button', () => {
//     it('should find next button via testId', () => {
//     const testIdName = 'nextButton';
//     const {getByTestId} = render(<CreateAccount1 />);
//     const foundButton = getByTestId(testIdName);
//     expect(foundButton).toBeTruthy();
//     });
  
//     it('Next button is pressable', () => {
//       const navigation = {navigate: () =>{}}
//       const page = render(<CreateAccount1 navigation={navigation}/>);
//       const nextButton = page.getByTestId('nextButton');
//       fireEvent.press(nextButton);
//     });
//   })


//   // Testing back button

// describe('Testing back button', () => {
//     it('should find back button via testId', () => {
//     const testIdName = 'backButton';
//     const {getByTestId} = render(<CreateAccount1 />);
//     const foundButton = getByTestId(testIdName);
//     expect(foundButton).toBeTruthy();
//     });
  
//     it('Back button is pressable', () => {
//       const navigation = {navigate: () =>{}}
//       const page = render(<CreateAccount1 navigation={navigation}/>);
//       const nextButton = page.getByTestId('backButton');
//       fireEvent.press(nextButton);
//     });

//     it('Back button navigates to login screen', () => {
//         const navigation = {navigate: () =>{}}
//         const page = render(<CreateAccount1 navigation={navigation}/>);
//         jest.spyOn(navigation, 'navigate')
//         const createAccount = page.getByTestId('backButton');
//         fireEvent.press(createAccount);
//         expect(navigation.navigate).toHaveBeenCalledWith("Login")
//     });
// })