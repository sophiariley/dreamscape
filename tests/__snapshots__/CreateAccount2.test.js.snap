// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "alignSelf": "center",
              "color": "#F8C98A",
              "fontSize": 35,
              "position": "absolute",
              "top": 100,
            }
          }
        >
          Create New Account
        </Text>
        <View
          style={
            {
              "alignSelf": "center",
              "position": "absolute",
              "top": 200,
            }
          }
        >
          <Text
            style={
              {
                "color": "#3A6496",
                "fontSize": 14,
                "paddingBottom": 5,
                "paddingTop": 10,
              }
            }
          >
            Enter Username
          </Text>
          <TextInput
            placeholder=""
            style={
              {
                "backgroundColor": "#DADADA",
                "borderRadius": 5,
                "fontSize": 20,
                "height": "20%",
                "paddingLeft": 10,
                "width": 350,
              }
            }
            testID="username"
          />
          <Text
            style={
              {
                "color": "#3A6496",
                "fontSize": 14,
                "paddingBottom": 5,
                "paddingTop": 10,
              }
            }
          >
            Enter Password
          </Text>
          <TextInput
            placeholder=""
            secureTextEntry={true}
            style={
              {
                "backgroundColor": "#DADADA",
                "borderRadius": 5,
                "fontSize": 20,
                "height": "20%",
                "paddingLeft": 10,
                "width": 350,
              }
            }
            testID="password"
          />
          <Text
            style={
              {
                "color": "#3A6496",
                "fontSize": 14,
                "paddingBottom": 5,
                "paddingTop": 10,
              }
            }
          >
            Re-enter Password
          </Text>
          <TextInput
            placeholder=""
            secureTextEntry={true}
            style={
              {
                "backgroundColor": "#DADADA",
                "borderRadius": 5,
                "fontSize": 20,
                "height": "20%",
                "paddingLeft": 10,
                "width": 350,
              }
            }
            testID="repPassword"
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "alignSelf": "center",
              "bottom": 80,
              "flex": 1,
              "flexDirection": "row",
              "height": "20%",
              "justifyContent": "center",
              "position": "absolute",
              "width": "70%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#D28A8E",
                "borderRadius": 5,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 15,
                "width": "100%",
              }
            }
            testID="nextButton"
          >
            <Text
              style={
                {
                  "color": "#F6F6F6",
                  "fontSize": 20,
                  "paddingRight": 10,
                }
              }
            >
              Next
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "alignSelf": "center",
              "bottom": 10,
              "flex": 1,
              "flexDirection": "row",
              "height": "20%",
              "justifyContent": "center",
              "position": "absolute",
              "width": "70%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#3A6496",
                "borderRadius": 5,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 15,
                "width": "100%",
              }
            }
            testID="backButton"
          >
            <Text
              style={
                {
                  "color": "#F6F6F6",
                  "fontSize": 20,
                  "paddingLeft": 10,
                }
              }
            >
              Previous
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
